src/featurestorebundle/delta/feature/FeaturesFilteringManager.py

```diff
@@ -19,34 +25,25 @@ def get_latest(
         id_column = feature_store.columns[0]
         time_column = feature_store.columns[1]
 
-        features_data = feature_store.join(rainbow_table, on=get_rainbow_table_hash_column().name).withColumn(
-            "features_intersect", f.array_intersect(get_rainbow_table_features_column().name, f.array(*map(f.lit, features)))
+        window = (
+            Window().partitionBy(id_column).orderBy(f.desc(time_column)).rowsBetween(Window.unboundedPreceding, Window.unboundedFollowing)
         )
 
-        relevant_features = features_data.select(f.array_distinct(f.flatten(f.collect_set("features_intersect")))).collect()[0][0]
-
         features_data = (
-            features_data.groupBy(id_column)
-            .agg(
-                *[
-                    f.array_max(
-                        f.collect_list(f.struct(f.array_contains("features_intersect", feature), f.col(time_column), f.col(feature)))
-                    )[feature].alias(feature)
-                    for feature in relevant_features
-                ],
-                f.array_distinct(f.flatten(f.collect_set("features_intersect"))).alias("computed_features"),
-            )
-            .filter(f.size("computed_features") == len(relevant_features))
-            .drop("computed_features")
+            feature_store.select(id_column, *[f.first(feature, ignorenulls=True).over(window).alias(feature) for feature in features])
```

Comment:
if you swap the comprehension from [] to () it should at least use much less memory compared to list

Reply:
you could even make it a `map(lambda feature: f.first(feature, ignorenulls=True).over(window).alias(feature)), features)` to make it super fast

Reply:
I think that in the end it's same. So I would rather prefer list comprehension as it is more pythonic
![image](https://user-images.githubusercontent.com/22962839/161004455-0a23403c-960a-4393-b970-9dfd830a0e85.png)